library/kotlin/src/io/envoyproxy/envoymobile/RequestHeaders.kt

```diff
@@ -0,0 +1,55 @@
+
+package io.envoyproxy.envoymobile
+
+/*
+ * Headers representing an outbound request.
+ */
+class RequestHeaders: Headers {
+  /**
+   * Internal constructor used by builders.
+   *
+   * @param headers: Headers to set.
+   */
+  internal constructor(headers: Map<String, List<String>>) : super(headers)
+
+  /**
+   * Method for the request.
+   */
+  val method: RequestMethod by lazy { RequestMethod.enumValue(value(":method")?.first()!!) }
+
+  /**
+   * The URL scheme for the request (i.e., "https").
+   */
+  val scheme: String by lazy { value(":scheme")?.first()!! }
+
+  /**
+   * The URL authority for the request (i.e., "api.foo.com").
+   */
+  val authority: String by lazy { value(":authority")?.first()!! }
+
+  /**
+   * The URL path for the request (i.e., "/foo").
+   */
+  val path: String by lazy { value(":path")?.first()!! }
+
+  /**
+   * Retry policy to use for this request.
+   */
+  val retryPolicy: RetryPolicy by lazy { RetryPolicy(this) }
```

Comment:
```suggestion
  val retryPolicy: RetryPolicy? by lazy { RetryPolicy(this) }
```
Shouldn't this be optional?

Reply:
Constructors cannot return null

Reply:
The constructor is not a "failable" constructor, no such concept in Kotlin afaict.

Reply:
@junr03 I think we should keep consistency here for this PR. It's a little weird/strange of a thing to do but we'll need to do something like this:
```suggestion
  val retryPolicy: RetryPolicy? by lazy {
    try {
      RetryPolicy(this)
    } catch (t: Throwable) {
      null
    }
  }
```

Reply:
Yea +1 to this. Another option would be to make a static function that returns an optional `ReturnPolicy?` instead of having a throwing constructor. Not sure which is more "acceptable" 🤷